autorest/adal/token.go

```diff
@@ -264,41 +259,57 @@ func NewServicePrincipalTokenFromCertificate(oauthConfig OAuthConfig, clientID s
 	)
 }
 
-// NewServicePrincipalTokenFromMSI creates a ServicePrincipalToken via the MSI VM Extension.
-func NewServicePrincipalTokenFromMSI(oauthConfig OAuthConfig, resource string, callbacks ...TokenRefreshCallback) (*ServicePrincipalToken, error) {
-	return newServicePrincipalTokenFromMSI(oauthConfig, resource, managedIdentitySettingsPath, callbacks...)
+// Resource enumerates resources from which MSI authentication is supported
+type Resource string
+
+const (
+	// VirtualMachine specifies MSI support for VMs
+	VirtualMachine Resource = "VM"
```

Comment:
I think golint should be flagging this. I think the name is supposed to be `ResourceVirtualMachine`

Separately, I think I do like this strategy better than trying to reflect over actual resources to figure out if they're supported. This will scale better.